### PR TITLE
Feat add three new injuries (black eye, heat stroke, dislocated jaw)

### PR DESCRIPTION
**Pull request by Darxo**
### Black Eye
image

### Heat Stroke
image

### Dislocated Jaw
**Not his turn**
image

**His Turn**
image

### CODE REVIEW

File: scripts/skills/injury/rf_black_eye_injury.nut
Comment (LordMidas) [inline scripts/skills/injury/rf_black_eye_injury.nut:52]: Please add hyperlinks in the tooltips for concepts such as vision, melee skill, ranged skill. Please go through the other files too and do the same where applicable.

File: scripts/skills/injury/rf_black_eye_injury.nut
Comment (LordMidas) [inline scripts/skills/injury/rf_black_eye_injury.nut:51]: Multiplicative changes should be without sign and use the words `increased/decreased` for now. Later we can change them to `more/less` in a single overall commit when we get to that point. Please go through the other files too and do the same where applicable.

File: scripts/skills/injury/rf_dislocated_jaw_injury.nut
Comment (LordMidas) [inline scripts/skills/injury/rf_dislocated_jaw_injury.nut:11]: `A forceful impact` instead of The

File: scripts/skills/injury/rf_heat_stroke_injury.nut
Comment (LordMidas) [inline scripts/skills/injury/rf_heat_stroke_injury.nut:10]: `Being exposed to intense heat caused this character\'s body temperature to soar to dangerous levels.`

File: scripts/skills/injury/rf_heat_stroke_injury.nut
Comment (LordMidas) [inline scripts/skills/injury/rf_heat_stroke_injury.nut:3]: Should be `FatigueEffectMult` because it applies to all fatigue built, not just from skills with fatigue cost.

File: scripts/skills/injury/rf_heat_stroke_injury.nut
Comment (LordMidas) [inline scripts/skills/injury/rf_heat_stroke_injury.nut:49]: This empty line before the last closing brackets should be removed. It's also present in the other files so please remove from them too.

File: scripts/skills/injury/rf_dislocated_jaw_injury.nut
Comment (LordMidas) [inline scripts/skills/injury/rf_dislocated_jaw_injury.nut:4]: Should we instead call this `BraveryMult`? At least that way in the code it is always called the same thing (Bravery instead of Resolve).

File: scripts/skills/injury/rf_black_eye_injury.nut
Comment (LordMidas) [inline scripts/skills/injury/rf_black_eye_injury.nut:23]: I would suggest changing this variable's name to `local ret =`. I've been doing that in my new files too. We do that with virtually all return values. What do you think?

File: scripts/skills/injury/rf_black_eye_injury.nut
Comment (LordMidas) [inline scripts/skills/injury/rf_black_eye_injury.nut:28]: I've been starting our ids at 10 (as it gives more room for other mods who may be adding between 7-9). What do you think?

**Review by LordMidas — CHANGES_REQUESTED** [review]

File: scripts/skills/injury/rf_black_eye_injury.nut
Comment (Darxo) [reply scripts/skills/injury/rf_black_eye_injury.nut:23]: I started that using `tooltip` because I saw someone else (you?) do it.
I like `ret`

File: scripts/skills/injury/rf_dislocated_jaw_injury.nut
Comment (Darxo) [reply scripts/skills/injury/rf_dislocated_jaw_injury.nut:4]: I don't like that for Bravery/Resolve in particular and would leave it up to the respective dev.

By that logic we would need to start using `Stamina` for some fatigue related variable names. Though on a side note I actually want that example because I still want to partly return to `Stamina`

File: scripts/skills/injury/rf_heat_stroke_injury.nut
Comment (Darxo) [reply scripts/skills/injury/rf_heat_stroke_injury.nut:3]: FatigueCostMult does not mention `skills`. I find it a very neutral term and always see it as including cost for moving around in it.
Using skills and moving around costs fatigue.

Other sources of fatigue build-up like getting hit, encumbrance or special triggered costs from skills are not influenced (en garde, rebuke)

File: scripts/skills/injury/rf_dislocated_jaw_injury.nut
Comment (LordMidas) [reply scripts/skills/injury/rf_dislocated_jaw_injury.nut:4]: In my opinion, when a multiplier applies to a certain property/field in an object it should conform to the naming scheme of THAT object. So, for example. if you want to modify the `_properties.Stamina` then you should use `StaminaMult` and if you want to modify the `actor.setFatigue` then you should use `FatigueModifier` etc.

This is the most stable and uniform way of working with these fields in my opinion and is the path of least resistance. Otherwise, everyone uses their own preferred style and it becomes a mess to browse through the codebase.

File: scripts/skills/injury/rf_heat_stroke_injury.nut
Comment (LordMidas) [reply scripts/skills/injury/rf_heat_stroke_injury.nut:3]: Same [comment as on the other conversation](https://github.com/Battle-Modders/mod-reforged/pull/292#discussion_r1566774650). If you are changing the `FatigueEffectMult` then this variable should be named similarly to make that clear.

File: scripts/skills/injury/rf_dislocated_jaw_injury.nut
Comment (LordMidas) [inline scripts/skills/injury/rf_dislocated_jaw_injury.nut:52]: `character\'s`

File: scripts/skills/injury/rf_heat_stroke_injury.nut
Comment (LordMidas) [inline scripts/skills/injury/rf_heat_stroke_injury.nut:29]: Need to invert color here to make it red.

File: scripts/skills/injury/rf_black_eye_injury.nut
Comment (LordMidas) [inline scripts/skills/injury/rf_black_eye_injury.nut:31]: We can take this opportunity to add `Vision` as a concept. To do this you can add a commit to the `z_nested_tooltips.nut` file and in the `AutoConcepts` array you can add `"character-stats.SightDistance"`. Then here it can be parsed using `[Vision|Concept.SightDistance]`

File: mod_reforged/hooks/config/character_injuries.nut
Comment (LordMidas) [inline mod_reforged/hooks/config/character_injuries.nut:1]: Need to add these injuries to the `msu_excluded_injuries.nut` file as well. I think all 3 of these injuries should be excluded from all kinds of Undead so you can simply add them to the `RF_Undead` excluded injuries list.

**Review by LordMidas — CHANGES_REQUESTED** [review]

File: scripts/skills/injury/rf_dislocated_jaw_injury.nut
Comment (Darxo) [reply scripts/skills/injury/rf_dislocated_jaw_injury.nut:4]: I'm ok with this way of writing those typical member variables now

File: mod_reforged/config/msu_excluded_injuries.nut
Comment (LordMidas) [inline mod_reforged/config/msu_excluded_injuries.nut:38]: Black Eye should be removed from RF_Undead instead of only from RF_Skeleton. Undead do not have a functioning blood circulatory system and therefore shouldn't be getting bruises or swollen body parts from getting hit.

Dislocated Jaw should also be removed from skeletons imo but we can discuss this in design discussion if you like.

File: scripts/skills/injury/rf_dislocated_jaw_injury.nut
Comment (LordMidas) [inline scripts/skills/injury/rf_dislocated_jaw_injury.nut:11]: just noticed a typo: `breathe`

**Review by LordMidas — CHANGES_REQUESTED** [review]

File: mod_reforged/hooks/config/character_injuries.nut
Comment (LordMidas) [inline mod_reforged/hooks/config/character_injuries.nut:2]: I'm sorry for missing this in my earlier reviews but I just noticed it now that these injuries should also additionally be added to the following vanilla arrays:
1. All three injuries go in `::Const.Injury.All`.
2. Black Eye and Dislocated Jaw go in `::Const.Injury.BluntAndPiercingHead`
3. Heat Stroke goes in `::Const.Injury.BurningAndPiercingBody` and `::Const.Injury.Burning`.

**Review by LordMidas — CHANGES_REQUESTED** [review]

File: mod_reforged/hooks/config/character_injuries.nut
Comment (LordMidas) [inline mod_reforged/hooks/config/character_injuries.nut:40]: Comma should be removed

File: mod_reforged/hooks/config/character_injuries.nut
Comment (LordMidas) [inline mod_reforged/hooks/config/character_injuries.nut:53]: Comma should be removed

File: mod_reforged/hooks/config/character_injuries.nut
Comment (LordMidas) [inline mod_reforged/hooks/config/character_injuries.nut:69]: Comma should be removed

File: mod_reforged/hooks/config/character_injuries.nut
Comment (LordMidas) [inline mod_reforged/hooks/config/character_injuries.nut:61]: Comma should be removed

**Review by LordMidas — APPROVED** [review]
Looks good to merge after the requested comma removals.

File: mod_reforged/hooks/config/character_injuries.nut
Comment (LordMidas) [reply mod_reforged/hooks/config/character_injuries.nut:40]: I am aware that commas are optional in squirrel (it will work even without commas in the entire array/table and with extra comma at the end). But your point that having commas at the end can lead to cleaner diff in commits is interesting. I don't know what the convention is in this case that we want to go with, so I'd ask Enduriel and if he's ok with this then we can do it.

Comment (LordMidas) [reply mod_reforged/hooks/config/character_injuries.nut:40]: Based on the discussion in Discord we decided that we like these commas at the end. So you can disregard this comment.

**Pull request merged**
